examples/multi_win.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use druid::shell::{runloop, WindowBuilder};
+use druid::widget::{ActionWrapper, Button, Column, DynLabel, Padding};
+
+use druid::{
+    Data, DruidHandler, Env, Event, EventCtxRoot, KeyCode, LayoutCtxRoot, PaintCtxRoot, RootWidget,
+    UpdateCtxRoot, Widget, WindowId, WindowSet,
+};
+
+pub struct SharedWindow<T: Data> {
+    windows: WindowSet<T>,
+}
+
+impl<T: Data> SharedWindow<T> {
+    pub fn new(root: impl Widget<T> + 'static, id: WindowId) -> SharedWindow<T> {
+        SharedWindow {
+            windows: WindowSet::new(root, id),
+        }
+    }
+}
+
+impl RootWidget<u32> for SharedWindow<u32> {
```

Comment:
All of the code in this example feels like code that should be provided by druid.

What if `RootWidget` wasn't a trait, but was a struct? It could, by default, implement a bunch of general behaviour, keep a stack of windows, handle system commands, etc; and then if the user wanted custom event handling, for instance, we could use a delegate pattern, like:

```rust
```rust
struct RootWidget<T> {
    ...
    event_delegate: Option<Box<dyn EventDelegate<T>>>,
}

trait EventDelegate<T> {
    /// Recieves an event before it is sent up the widget tree.
    /// Can return either the original event or a new event to be passed
    /// up the tree (for instance, replacing a key press with a command),
    /// or can return `None` to swallow this event.
    fn event(&mut self, event: Event, ctx: &mut dyn WinCtx) -> Option<Event>;
}
```

I guess my question is: what logic, exactly, do we expect to want at the application level? Do we ever imagine the user wanting to override `paint`? `layout`? What would that mean?

I'm also thinking about this in terms of menus. Currently, if you ever want to modify a menu you will need to do it manually in a widget. It might be nice if, instead, there was a `MenuDelegate` trait, where we would call you on `update` and you could request to change a menu.